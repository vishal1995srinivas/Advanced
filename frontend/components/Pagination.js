import React from 'react';
import PaginationStyles from './styles/PaginationStyles';
import gql from 'graphql-tag';
import { Query } from 'react-apollo';
import perPage from '../config';
const PAGINATION_QUERY = gql`
	query PAGINATION_QUERY {
		itemsConnection {
			aggregate {
				count
			}
		}
	}
`;

const Pagination = (props) => (
	<PaginationStyles>
		<Query query={PAGINATION_QUERY}>
			{({ data, loading, error }) => {
				if (loading) return <p>Loading...</p>;
				const count = data.itemsConnection.aggregate.count;
				const pages = Math.ceil(count / 4); //perPage is not loading from config
				return (
					<p>
						Page {props.page} of {pages}
					</p>
				);
			}}
		</Query>
	</PaginationStyles>
);

export default Pagination;
