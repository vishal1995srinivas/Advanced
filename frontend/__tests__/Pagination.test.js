import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import Router from 'next/router';
import Pagination, { PAGINATION_QUERY } from '../components/Pagination';
import { MockedProvider } from 'react-apollo/test-utils';

function makeMocksFor(length) {
	return [
		{
			request: { query: PAGINATION_QUERY },
			result: {
				data: {
					itemsConnection: {
						__typename: 'aggregate',
						aggregate: {
							__typename: 'count',
							count: length
						}
					}
				}
			}
		}
	];
}
describe('<Pagination/>', () => {
	it('displays a loading message', () => {
		const wrapper = mount(
			<MockedProvider mocks={makeMocksFor(1)}>
				<Pagination page={1} />
			</MockedProvider>
		);
		const pagination = wrapper.find('[data-test="pagination"]');
		expect(wrapper.text()).toContain('Loading...');
	});
});
