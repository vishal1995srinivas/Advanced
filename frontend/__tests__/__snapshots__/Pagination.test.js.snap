// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> renders a pagination for 18 items 1`] = `
<div
  className="PaginationStyles-aduuar-0 ewJsCc"
  data-test="pagination"
>
  <SideEffect(Head)>
    <Head />
  </SideEffect(Head)>
  <Link
    href={
      Object {
        "pathname": "Items",
        "query": Object {
          "page": 0,
        },
      }
    }
    prefetch={true}
  >
    <a
      aria-disabled={true}
      className="prev"
      href="Items?page=0"
      onClick={[Function]}
    >
      ⬅ Prev
    </a>
  </Link>
  <p>
    Page 
    1
     of 
    <span
      className="totalPages"
    >
       
      5
    </span>
  </p>
  <p>
    18
     Items Total
  </p>
  <Link
    href={
      Object {
        "pathname": "Items",
        "query": Object {
          "page": 2,
        },
      }
    }
    prefetch={true}
  >
    <a
      aria-disabled={false}
      className="next"
      href="Items?page=2"
      onClick={[Function]}
    >
      Next ➡
    </a>
  </Link>
</div>
`;
